pkg/controller/clientutil/delete.go

```diff
@@ -0,0 +1,35 @@
+package clientutil
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func DeleteYaml(ctx context.Context, c client.Client, yaml []byte) error {
```

Comment:
Do these need to exactly match to be deleted? Or is it based on the resource kind and name? I know you are adding this tooling but what is an example in which this is used?

Reply:
It will be used in the cilium reconciler, to delete the preflight objects we generate with helm.

It should only need to match resource type (kind and api version), name and namespace